src/main/java/com/exadel/discount/service/impl/DiscountServiceImpl.java

```diff
@@ -112,6 +113,17 @@ public DiscountDTO restoreById(UUID id) {
         return discountMapper.getDTO(discount);
     }
 
+    @Override
+    public List<DiscountDTO> search(Integer size, String searchText) {
+        log.debug("Getting list of all Discounts by searchText");
+        List<Discount> discounts = discountRepository
+                .findAll(prepareSearchPredicate(searchText), Sort.by("viewNumber"));
+        Page<Discount> discountPage = PageableExecutionUtils
```

Comment:
Paging should happen on the DB call, not after

Reply:
+